javascripts/discourse/initializers/search-banner.js

```diff
@@ -6,15 +6,20 @@ export default {
   initialize() {
     withPluginApi("1.1.0", (api) => {
       const site = api.container.lookup("service:site");
-      api.reopenWidget("header", {
-        didRenderWidget() {
-          if (!site.isMobileDevice) {
-            document
-              .querySelector(".d-header")
-              .classList.add("search-header--visible");
-          }
-        },
-      });
+      const currentUser = api.getCurrentUser();
+      if (currentUser?.glimmer_header_enabled && !site.isMobileDevice) {
```

Comment:
Hmm, I'm not sure we want to start introducing `currentUser?.glimmer_header_enabled` calls in loads of themes and plugins. That'll make it difficult for us to drop the flag in future.

Could we leave both `addCustomHeaderClass` and `reopenWidget` calls in place, without an if statement? And then when we drop the header widget, it'll start printing a warning to the console? :thinking: